File: demo/KSYLiveDemo/KSYDemoUI/KSYPipView.m
//
//  KSYPipView.m
//  KSYGPUStreamerDemo
//
//  Created by 孙健 on 16/6/24.
//  Copyright © 2016年 ksyun. All rights reserved.
//

#import "KSYPipView.h"
#import "KSYNameSlider.h"
#import "KSYFileSelector.h"

@interface KSYPipView (){
    UILabel * _pipTitle;
    KSYFileSelector * _pipSel;
    KSYFileSelector * _bgpSel;
}
@end

@implementation KSYPipView
-(id)init{
    self = [super init];
    _pipStatus  = @"idle";
    _pipTitle   = [self addLable:@"画中画地址 Documents/movies"];
    _pipTitle.numberOfLines = 2;
    _pipTitle.textAlignment = NSTextAlignmentLeft;
    
    _progressV  = [[UIProgressView alloc] init];
    [self addSubview:_progressV];
    _pipPlay    = [self addButton:@"播放"];
    _pipPause   = [self addButton:@"暂停"];
    _pipStop    = [self addButton:@"停止"];
    _pipNext    = [self addButton:@"下一个视频文件"];
    _bgpNext    = [self addButton:@"下一个背景图片"];
    _volumSl    = [self addSliderName:@"音量" From:0 To:100 Init:50];
    
    _pipPattern = @[@".mp4", @".flv"];
    _bgpPattern = @[@".jpg",@".jpeg", @".png"];
    
    _pipSel = [[KSYFileSelector alloc] initWithDir:@"/Documents/movies/"
                                         andSuffix:_pipPattern];
    _bgpSel = [[KSYFileSelector alloc] initWithDir:@"/Documents/images/"
                                         andSuffix:_bgpPattern];
    if(_pipSel.filePath){
        _pipURL = [NSURL fileURLWithPath:_pipSel.filePath];
    }
    if(_bgpSel.filePath){
        _bgpURL = [NSURL fileURLWithPath:_bgpSel.filePath];
    }
    return self;
}
- (void)layoutUI{
    [super layoutUI];
    self.btnH = 10;
    [self putRow1:_progressV];
    self.btnH = 60;
    [self putRow1:_pipTitle];
    self.btnH = 30;
    [self putRow:@[_pipPlay,_pipPause, _pipStop] ];
    [self putRow1:_volumSl];
    [self putRow2:_pipNext
              and:_bgpNext];
}

- (IBAction)onBtn:(id)sender {
    if (sender == _pipNext){
        if( [_pipSel selectFileWithType:KSYSelectType_NEXT]){
            _pipURL = [NSURL fileURLWithPath:_pipSel.filePath];
        }
    }
    if (sender == _bgpNext){
        if( [_bgpSel selectFileWithType:KSYSelectType_NEXT] ){
            _bgpURL = [NSURL fileURLWithPath:_bgpSel.filePath];
        }
    }
    _pipTitle.text = [NSString stringWithFormat:@"%@: %@\n%@", _pipStatus, _pipSel.fileInfo, _bgpSel.fileInfo ];
    [super onBtn:sender];
}
@end

File: demo/KSYLiveDemo_Swift/KSYDemoUI_Swift/subviews/KSYPipView.swift
//
//  KSYPipView.swift
//  KSYLiveDemo_Swift
//
//  Created by iVermisseDich on 17/1/10.
//  Copyright © 2017年 com.ksyun. All rights reserved.
//

import UIKit

class KSYPipView: KSYUIView {

    var pipPlay: UIButton?
    var pipPause: UIButton?
    var pipStop: UIButton?
    var progressV: UIProgressView?
    var volumSl: KSYNameSlider?
    var pipNext: UIButton?
    var bgpNext: UIButton?
    
    // 当前画中画的视频和背景图片的路径
    var pipURL: URL?
    var bgpURL: URL?
    // 当前画中画的播放状态
    var pipStatus: String?
    
    // match pattern
    var pipPattern: [String]?
    var bgpPattern: [String]?
    
    private
    var _pipTitle: UILabel?
    var _pipSel: KSYFileSelector?
    var _bgpSel: KSYFileSelector?
    
    required init?(coder aDecoder: NSCoder) {
        fatalError("init(coder:) has not been implemented")
    }
    
    override init(withParent pView: KSYUIView) {
        super.init(withParent: pView)
        pipStatus = "idle"
        _pipTitle = addLabel(title: "")
        _pipTitle?.numberOfLines = 2
        _pipTitle?.textAlignment = .left
        
        progressV = UIProgressView()
        addSubview(progressV!)
        pipPlay = addButton(title: "播放")
        pipPause = addButton(title: "暂停")
        pipStop = addButton(title: "停止")
        pipNext = addButton(title: "下一个视频文件")
        bgpNext = addButton(title: "下一个背景图片")
        volumSl = addSlider(name: "音量", from: 0, to: 100, initV: 50)
        
        pipPattern = [".mp4", ".flv"]
        bgpPattern = [".jpg", ".jpeg", ".png"]
        
        _pipSel = KSYFileSelector.init(dir: "/Documents/movies/", suf: pipPattern!)
        _bgpSel = KSYFileSelector.init(dir: "/Documents/images/", suf: bgpPattern!)
        
        if let filePath = _pipSel?.filePath , filePath.characters.count > 0{
            pipURL = URL.init(fileURLWithPath: _pipSel!.filePath)
        }
        if let filePath = _bgpSel?.filePath , filePath.characters.count > 0 {
            bgpURL = URL.init(fileURLWithPath: _bgpSel!.filePath)
        }
    }
    
    override func layoutUI() {
        super.layoutUI()
        btnH = 10
        putRow1(subV: progressV!)
        btnH = 60
        putRow1(subV: _pipTitle)
        btnH = 30
        putRow(subV: [pipPlay!, pipPause!, pipStop!])
        putRow1(subV: volumSl)
        putRow2(subV0: pipNext, and: bgpNext)
    }
    
    override func onBtn(sender: AnyObject) {
        if sender as? NSObject == pipNext {
            guard let _ = _pipSel else {
                return
            }
            if _pipSel!.selectFileWithType(type: .NEXT) {
                pipURL = URL.init(fileURLWithPath: _pipSel!.filePath)
            }
        }else if sender as? NSObject == bgpNext {
            guard let _ = _bgpSel else {
                return
            }
            if _bgpSel!.selectFileWithType(type: .NEXT) {
                bgpURL = URL.init(fileURLWithPath: _bgpSel!.filePath)
            }
        }
        
        _pipTitle?.text = "\(pipStatus ?? "idle"): \(_pipSel!.fileInfo)\n\(_bgpSel!.fileInfo)"
        super.onBtn(sender: sender)
    }

}
